### DISCUSSION TITLE
regexManager not working on Azure/azure-cli with github-tags

### DISCUSSION THREAD

**Discussion by eana** [renovatebot/renovate#21325] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.23.4

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab 15.9.4

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want to update the `ENV` variable defined in a `Dockerfile` with the latest github-tag fetched from the [azure-cli](https://github.com/Azure/azure-cli) repository

### What you tried so far.

Hi,

I am currently using the regex manager to capture advanced matches for many `ENV` variables that need to be kept up-to-date. Most of these are working well with our approach, which is also used in other places. However, one of them is causing problems.

In my `Dockerfile`, I have the following:

```
# renovate: datasource=github-tags depName=Azure/azure-cli versioning=loose
ENV AZCLI_VERSION="azure-cli-2.46.0"
```

Here is the configuration for `Renovate`:

```json
  "regexManagers": [
    {
      "fileMatch": ["(^|/)Dockerfile$"],
      "matchStrings": [
        "#\\s*renovate:\\s*datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=\"?(?<currentValue>.*?)\"?\\s"
      ],
      "versioningTemplate": "{{#if versioning}}{{versioning}}{{else}}semver{{/if}}"
    }
  ],
```

However, `Renovate` fails to update the version and returns a `skipReason` of `invalid-value`. Can you help me figure out what I'm doing wrong?

Thanks.

### Relevant debug logs

<details><summary>Logs</summary>

```
         "regex": [
           {
             "deps": [
               {
                 "depName": "Azure/azure-cli",
                 "currentValue": "azure-cli-2.46.0",
                 "datasource": "github-tags",
                 "versioning": "loose",
                 "replaceString": "# renovate: datasource=github-tags depName=Azure/azure-cli versioning=loose\nENV AZCLI_VERSION=\"azure-cli-2.46.0\"\n",
                 "updates": [],
                 "packageName": "Azure/azure-cli",
                 "warnings": [],
                 "skipReason": "invalid-value"
               }
             ],
             "matchStrings": [
               "#\\s*renovate:\\s*datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=\"?(?<currentValue>.*?)\"?\\s"
             ],
             "versioningTemplate": "{{#if versioning}}{{versioning}}{{else}}semver{{/if}}",
             "packageFile": "Dockerfile"
           }
         ]
```

</details>

**Comment by secustor** [comment]
You should not match `azure-cli-2.46.0` as currentValue but `2.46.0`. Alternatively  you can add an `extractVersionTemplate` option like you do with `versioning` to strip `azure-cli-`.

**Reply by rarkins** [reply]
Yes, match `2.46.0` as currentValue, and then use `extractVersion` for the datasource to strip out the prefix

**Reply by eana** [reply, marked as answer]
That did the trick, thank you very much. In case other will have the same issue this is what I did:

`Dockerfile`:

```
# renovate: datasource=github-tags depName=Azure/azure-cli versioning=loose
ENV AZCLI_VERSION="2.46.0"
```

`renovate.json`:

```json
  "regexManagers": [
    {
      "fileMatch": ["(^|/)Dockerfile$"],
      "matchStrings": [
        "#\\s*renovate:\\s*datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=\"?(?<currentValue>.*?)\"?\\s"
      ],
      "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
    }
  ],
  "packageRules": [
    {
        "packagePatterns": ["^Azure\\/azure-cli"],
        "extractVersion": "^azure-cli-(?<version>.*)$"
    }
  ],
```
